### PR TITLE
CASSANDRA-19577 Make queries visible to the "system_views.queries" virtual table at the coordinator level

### PR DESCRIPTION
**Pull request by maedhroz**
patch by Caleb Rackliffe; reviewed by Chris Lohfink for CASSANDRA-19577

### CODE REVIEW

File: src/java/org/apache/cassandra/concurrent/TaskFactory.java [hunk 1]
```diff
@@ -90,6 +90,10 @@ public Runnable toExecute(WithResources withResources, Runnable runnable)
         @Override
         public <T> RunnableFuture<T> toSubmit(WithResources withResources, Runnable runnable)
         {
+            if (runnable instanceof RunnableDebuggableTask)
+                return withResources.isNoOp() ? newTask(runnable)
+                                              : newTask(withResources, runnable);
+
             return withResources.isNoOp() ? newTask(callable(runnable))
                                           : newTask(withResources, callable(runnable));
         }
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/concurrent/TaskFactory.java]: is there a reason to keep this logic and guard the above logic with `if (runnable instanceof RunnableDebuggableTask)`?  As far as I can tell there isn't a good reason... we convert the runnable to `callable` for the execution logic, and the debug logic deals with the raw type...

File: test/distributed/org/apache/cassandra/distributed/test/QueriesTableTest.java [hunk 2]
```diff
@@ -18,72 +18,152 @@
...
 
+    private static Cluster SHARED_CLUSTER;
+    private static com.datastax.driver.core.Cluster DRIVER_CLUSTER;
+    private static Session SESSION;
+
+    @BeforeClass
+    public static void createCluster() throws IOException
+    {
+        SHARED_CLUSTER = init(Cluster.build(1).withConfig(c -> c.with(Feature.NATIVE_PROTOCOL)).start());
+        DRIVER_CLUSTER = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/QueriesTableTest.java]: FYI `org.apache.cassandra.distributed.test.JavaDriverUtils#create(org.apache.cassandra.distributed.api.ICluster<? extends org.apache.cassandra.distributed.api.IInstance>)`

does the same thing in this case.

File: test/distributed/org/apache/cassandra/distributed/test/QueriesTableTest.java [hunk 3]
```diff
@@ -18,72 +18,152 @@
...
+        {
+            for (int i = 0; i < ITERATIONS; i++)
+            {
+                SESSION.executeAsync("INSERT INTO " + KEYSPACE + ".tbl (k, v) VALUES (" + i + ", 0)");
+                SESSION.executeAsync("SELECT * FROM " + KEYSPACE + ".tbl WHERE k = " + (i - 1));
+            }
+
+            totalOperations += ITERATIONS;
+
+            SimpleQueryResult result = SHARED_CLUSTER.get(1).executeInternalWithResult("SELECT * FROM system_views.queries");
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/QueriesTableTest.java]: this is super racy and leaves you only validating that you saw reads/writes and not that they were "correct".  Might want to use BB to make this test stable and avoid timing out (keep running over and over trying to find the last query)

**Review by dcapwell — APPROVED** [review]

File: src/java/org/apache/cassandra/concurrent/TaskFactory.java (hunk 1 above)
Comment (maedhroz) [reply src/java/org/apache/cassandra/concurrent/TaskFactory.java]: It certainly doesn't fail any tests to simplify this...

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/428fa1f4453735ef43d508621e5e7d9e0a054415

**Pull request closed without merging**